hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hudi/ProvidesHoodieConfig.scala

```diff
@@ -174,7 +174,7 @@ trait ProvidesHoodieConfig extends Logging {
 
     val enableHive = isUsingHiveCatalog(sparkSession)
 
-    withSparkConf(sparkSession, catalogProperties) {
+    withSparkConf(sparkSession, catalogProperties ++ extraOptions) {
```

Comment:
Could you add a test that should fail before this fix, to guard any breaking changes in the future?

Reply:
I can't figure out how to implement the test case.Do you have any ideas?